libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -33,18 +33,27 @@ void VotePacketHandler::process(const PacketData &packet_data, const std::shared
     return;
   }
 
-  // Adds unverified vote into local structure
-  if (vote_mgr_->voteInVerifiedMap(vote) || !vote_mgr_->addUnverifiedVote(vote)) {
-    LOG(log_dg_) << "Received PBFT vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
-                 << ") already saved in (un)verified queues.";
+  // No need to check unverified map as it is checked inside addUnverifiedVote
+  if (vote_mgr_->voteInVerifiedMap(vote)) {
+    LOG(log_dg_) << "Vote " << vote_hash << " (from " << packet_data.from_node_id_.abridged()
+                 << ") already saved in verified queue";
+    return;
+  }
+  auto vote_ptr_copy = vote;
```

Comment:
I think could remove the "vote_ptr_copy", and remove "std::move" in "if (!vote_mgr_->addUnverifiedVote(std::move(vote)))"

Reply:
no you cant, addUnverifiedVote has an argument of type `std::shared_ptr<Vote>&&`, which means you gotta use std::move

Reply:
I think the purpose of using std::move is for saving space for passing parameters. Here in order to use std::move, have to make a copy first. 
If like that, I think we should use reference "std::shared_ptr<Vote>&" that should be better

Reply:
First of all, when you comment on something and request change, you should understand the topic you write about. Move semantics does not really save space for passing parameters, it's purpose is completely different. Plus you are missing big picture here. I made argument of `addUnverifiedVote` method `Vote&&` so I would optimize `addUnverifiedVotes` function, that is important in terms of performance as it pushes hundreds of votes and does it regurarly inside pbft_manager. It is ok that we make copies in packet handlers as long as the performance in some more critical code(pbft manager) is optimized thanks to it... There are always other ways how to do it, but you are clearly not suggesting a new way, just commenting on one line without considering the bigger picture. I guess I can consider this conversation finished